include/freerdp/json.h

```diff
@@ -0,0 +1,64 @@
+/**
+ * FreeRDP: A Remote Desktop Protocol Implementation
+ * Client Interface
+ *
+ * Copyright 2024 Ondrej Holy <oholy@redhat.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FREERDP_JSON_H
+#define FREERDP_JSON_H
+
+#if defined(WITH_CJSON)
+
+#include <cjson/cJSON.h>
+
+#if CJSON_VERSION_MAJOR == 1
+#if CJSON_VERSION_MINOR <= 7
+#if CJSON_VERSION_PATCH < 13
+#define USE_CJSON_COMPAT
+#endif
+#endif
+#endif
+
+#elif defined(WITH_JSONC)
+
+#include <json-c/json.h>
```

Comment:
not sure this is an `API` we´d like to ship as our own like this in public headers.
might be better to use some thin wrapper and use our own naming scheme here to have that abstracted.

Reply:
other option would be to just move this to internals like before with `cJSON` so that we do not expose this ourselves. (you need to redo the used symbols for `SDL` client, but well...)

Reply:
Well, I expected that you wouldn't like this approach. I am just preparing a wrapper to avoid the code duplication...